lib/mix/lib/mix/release.ex

```diff
@@ -263,6 +270,21 @@ defmodule Mix.Release do
     |> Keyword.put(:kernel, :permanent)
   end
 
+  defp validate_steps!(steps) do
+    if not is_list(steps) or Enum.any?(steps, &(&1 != :assemble and not is_function(&1, 1))) do
+      Mix.raise(
+        "The :steps option must be a list of anonymous functions that receive one argument " <>
```

Comment:
To remove ambiguity, what do you think about?

```
The :steps option must be a list of:

  * anonymous function that receives one argument
  * the atom :assemble

Got: #{inspect(steps)}
```

Reply:
Excellent! :+1: